### PR TITLE
8270996: javadoc: missing comments in serialized classes

### PR DESCRIPTION
**Pull request by andy-goryachev-oracle**
Adding `@SuppressWarnings("doclint:missing")` to the lines in Serializable classes that generated javadoc's "missing comments" warning.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8270996](https://bugs.openjdk.org/browse/JDK-8270996): javadoc: missing comments in serialized classes (**Bug** - P4)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1386/head:pull/1386` \
`$ git checkout pull/1386`

Update a local copy of the PR: \
`$ git checkout pull/1386` \
`$ git pull https://git.openjdk.org/jfx.git pull/1386/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1386`

View PR using the GUI difftool: \
`$ git pr show -t 1386`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1386.diff">https://git.openjdk.org/jfx/pull/1386.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1386#issuecomment-1971599436)

### CODE REVIEW

File: modules/javafx.base/src/main/java/javafx/event/Event.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
  * to events of the same {@code Event} class.
  * @since JavaFX 2.0
  */
+@SuppressWarnings("doclint")
 public class Event extends EventObject implements Cloneable {
```
Comment (kevinrushforth) [inline modules/javafx.base/src/main/java/javafx/event/Event.java:45]: I don't think we should suppress all warnings, especially at the class level.

**Review by kevinrushforth — CHANGES_REQUESTED** [review]
I am not in favor of suppressing all javadoc warnings, especially at the class level. Is there no way to suppress just the serial warnings?

If we want to suppress the warnings (rather than providing actual docs for the fields, which may have little value), I think they should be moved down to the individual elements.

File: modules/javafx.base/src/main/java/javafx/event/Event.java (hunk 1 above)
Comment (andy-goryachev-oracle) [reply modules/javafx.base/src/main/java/javafx/event/Event.java]: for some reason I though it did not work -- corrected, fixed by suppressing the offending lines only with `@SuppressWarnings("doclint:missing")`

**Comment by andy-goryachev-oracle** [comment]
@arapte or @prsadhuk  could you please take a look?

File: modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java [hunk 2]
```diff
@@ -135,6 +135,14 @@
  *         }
  *     }
  * </pre>
+ * <strong>Warning:</strong>
+ * Serialized objects of this class will not be compatible with
+ * future Swing releases. The current serialization support is
```
Comment (kevinrushforth) [inline modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:140]: I would reword this to add "or future JavaFX releases".

File: modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java [hunk 3]
```diff
@@ -135,6 +135,14 @@
...
+ * appropriate for short term storage or RMI between applications running
+ * the same version of Swing.  As of 1.4, support for long term storage
```
Comment (kevinrushforth) [inline modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:142]: Add "and the same version of JavaFX" after Swing.

I would remove everything after the "As of 1.4 ..." since that isn't really relevant for JavaFX components and could be confusing.

**Review by kevinrushforth — COMMENTED** [review]
This now looks good to me, except for a couple of comments on the added docs to `JFXPanel`.

@prsadhuk Can you also review the changes to `JFXPanel`?

**Review by kevinrushforth — APPROVED** [review]
Looks good.

File: modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java [hunk 4]
```diff
@@ -135,6 +135,11 @@
  *         }
  *     }
  * </pre>
+ * <strong>Warning:</strong>
+ * Serialized objects of this class will not be compatible with
+ * future Swing or JavaFX releases. The current serialization support is
+ * appropriate for short term storage or RMI between applications running
+ * the same version of Swing and the same version of JavaFX.
```
Comment (prsadhuk) [inline modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:142]: Shouldn;t we need to add
@SuppressWarnings("serial") // Same-version serialization only

which I see in almost all classes in Swing which has the above serialization warning...

Comment (kevinrushforth) [reply modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:142]: This PR is addressing doclint warnings, not javac lint warnings. javac warnings can be addressed in a follow-up bug.

Comment (andy-goryachev-oracle) [reply modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:142]: FYI: after merging this and #1384 together all I see in the build log are 4 native warnings (on macOS):

```
/Users/angorya/Projects/jfx-1/jfx/rt/modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:93:21: warning: 'lockFocusIfCanDraw' is deprecated: first deprecated in macOS 10.14 - To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view. [-Wdeprecated-declarations]
        if ([nsView lockFocusIfCanDraw] == NO) {
                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSView.h:190:1: note: 'lockFocusIfCanDraw' has been explicitly marked deprecated here
- (BOOL)lockFocusIfCanDraw API_DEPRECATED("To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view.", macos(10.0,10.14));
^
/Users/angorya/Projects/jfx-1/jfx/rt/modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:98:25: warning: 'unlockFocus' is deprecated: first deprecated in macOS 10.14 - To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view. [-Wdeprecated-declarations]
                [nsView unlockFocus];
                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSView.h:189:1: note: 'unlockFocus' has been explicitly marked deprecated here
- (void)unlockFocus API_DEPRECATED("To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view.", macos(10.0,10.14));
^
/Users/angorya/Projects/jfx-1/jfx/rt/modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:120:24: warning: 'setView:' is deprecated: first deprecated in macOS 10.14 - Use NSOpenGLView to provide OpenGL content in a Cocoa app. [-Wdeprecated-declarations]
            [nsContext setView : nsView];
                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:193:36: note: property 'view' is declared deprecated here
@property (nullable, weak) NSView *view NS_SWIFT_UI_ACTOR API_DEPRECATED("", macos(10.0,10.14));
                                   ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:194:1: note: 'setView:' has been explicitly marked deprecated here
- (void)setView:(nullable NSView *)view NS_SWIFT_UI_ACTOR API_DEPRECATED("Use NSOpenGLView to provide OpenGL content in a Cocoa app.", macos(10.0,10.14));
^
/Users/angorya/Projects/jfx-1/jfx/rt/modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:121:21: warning: 'unlockFocus' is deprecated: first deprecated in macOS 10.14 - To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view. [-Wdeprecated-declarations]
            [nsView unlockFocus];
                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSView.h:189:1: note: 'unlockFocus' has been explicitly marked deprecated here
- (void)unlockFocus API_DEPRECATED("To draw, subclass NSView and implement -drawRect:; AppKit's automatic deferred display mechanism will call -drawRect: as necessary to display the view.", macos(10.0,10.14));
^
4 warnings generated.
```


as well as a number of these messages:

```
-do-compile:
    [mkdir] Created dir: /Users/angorya/Projects/jfx-1/jfx/rt/apps/toys/PickTest3D/build/empty
    [mkdir] Created dir: /Users/angorya/Projects/jfx-1/jfx/rt/apps/toys/PickTest3D/build/generated-sources/ap-source-output
    [javac] Compiling 6 source files to /Users/angorya/Projects/jfx-1/jfx/rt/apps/toys/PickTest3D/build/classes
    [javac] warning: [options] system modules path not set in conjunction with -source 9
    [javac] Note: /Users/angorya/Projects/jfx-1/jfx/rt/apps/toys/PickTest3D/src/picktest/PickTest3D.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 1 warning
```

Comment (kevinrushforth) [reply modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:142]: That's because we don't enable javac lint warnings by default. I am getting ready to file a few JBS issues related to this. Stay tuned.

In the mean time, if you want to see how many warnings we do have, try running with `gradle -PLINT=all`. Or rather, you'll see how many modules hit the 100 warnings limit. :)

Comment (prsadhuk) [reply modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:142]: So, that
@SuppressWarnings("serial") // Same-version serialization only
would be addressed as javac lint warning JBS?

Just thinking out aloud, I dont see this @SuppressWarnings("doclint:missing") in our java.desktop classes? what's the difference there?

**Review by prsadhuk — APPROVED** [review]
JFXPanel change ok to me..

File: modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java (hunk 4 above)
Comment (kevinrushforth) [reply modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java:142]: > So, that @SuppressWarnings("serial") // Same-version serialization only would be addressed as javac lint warning JBS?

Yes. I filed [JDK-8327170](https://bugs.openjdk.org/browse/JDK-8327170) as an umbrella task to address javac lint warnings, and will file new linked blocking bugs to address individual warnings.

> Just thinking out aloud, I dont see this @SuppressWarnings("doclint:missing") in our java.desktop classes? what's the difference there?

We wanted to silence the warnings (so we can turn on `-Werror` once the last warning is gone), and didn't see any real benefit in actually documenting the private fields. We could revisit that later if there was a reason, but I suspect there isn't.
